digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadwayChangesDAO.scala

```diff
@@ -594,12 +594,13 @@ SELECT
                    |	new_administrative_class
                    |FROM roadway_changes rc
                    |JOIN project p ON rc.project_id = p.id
-                   |-- Get the valid road name for the road that was modified in the project
-                   |LEFT JOIN road_name rn ON (rc.new_road_number = rn.road_number OR  rc.old_road_number = rn.road_number)
+                   |-- Get the valid road name for the road that was modified in the project prioritizing the new road number
+                   |LEFT JOIN road_name rn ON (rc.new_road_number = rn.road_number OR  (rc.new_road_number is null and rc.old_road_number = rn.road_number))
```

Comment:
Olisko kompaktimpi näin?
left join road_name rn on (rn.road_number = coalesce(rc.new_road_number, rc.old_road_number)